openstack/keppel/files/proxy.conf.template

```diff
@@ -0,0 +1,42 @@
+resolver ${NGINX_LOCAL_RESOLVERS};
+
+real_ip_header proxy_protocol;
+set_real_ip_from 0.0.0.0/0; # TODO: set this correct!!!!
+
+server {
+    listen 80 default_server proxy_protocol;
+    return 301 https://$host$request_uri;
+}
+
+server {
+  listen 443 ssl proxy_protocol;
+  server_name  keppel.qa-global.cloud.sap;
+
+  ssl_certificate /etc/nginx/ssl/tls.crt;
+  ssl_certificate_key /etc/nginx/ssl/tls.key;
+
+  # https://ssl-config.mozilla.org/#server=nginx&config=intermediate
+  ssl_protocols TLSv1.2 TLSv1.3;
+  ssl_ecdh_curve X25519:prime256v1:secp384r1;
+  ssl_ciphers ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:DHE-RSA-AES128-GCM-SHA256:DHE-RSA-AES256-GCM-SHA384:DHE-RSA-CHACHA20-POLY1305;
+  ssl_prefer_server_ciphers off;
+
+  ssl_session_timeout 1d;
+  ssl_session_cache shared:MozSSL:10m;
+
+  gzip on;
+  keepalive_timeout  65;
+
+  location / {
+    proxy_pass http://keppel-api.keppel.svc:80;
+    proxy_set_header Host            $host;
+    proxy_set_header X-Real-IP       $proxy_protocol_addr;
+    proxy_set_header X-Forwarded-For $proxy_protocol_addr;
+  }
+
+  # redirect server error pages to the static page /50x.html
+  error_page 500 502 503 504  /50x.html;
+  location = /50x.html {
+    root /usr/share/nginx/html;
+  }
```

Comment:
I don't want error responses to be obscured on this level. Besides, HTML documents will confuse API users like `docker pull` who expect a structured error response.